Sources/NIOPosix/DatagramVectorReadManager.swift

```diff
@@ -178,7 +178,9 @@ struct DatagramVectorReadManager {
 #else
             precondition(self.messageVector[i].msg_hdr.msg_namelen != 0, "Unexpected zero length peer name")
 #endif
-            let address: SocketAddress = self.sockaddrVector[i].convert()
+            let address: SocketAddress
+            do { address = try self.sockaddrVector[i].convert() }
+            catch { fatalError("Socket address conversion failed.") }
```

Comment:
Prefer `try!`.

Reply:
Good to know—I assumed that `try!` would be discouraged in review. Although I appreciate that all we do here is fail with a different string.

Changed in 805f1ca82.

Reply:
`try!` falls into the same bucket as `!`, which is that you need to write down why you know it won't crash in a comment.